ui/dashboards/src/views/ViewDashboard/DashboardApp.tsx

```diff
@@ -108,7 +109,12 @@ export const DashboardApp = (props: DashboardAppProps) => {
       />
       <Box sx={{ padding: (theme) => theme.spacing(2), height: '100%' }}>
         <ErrorBoundary FallbackComponent={ErrorAlert}>
-          <Dashboard emptyDashboard={emptyDashboard} />
+          <Dashboard
+            emptyDashboardProps={{
+              onEditButtonClick,
```

Comment:
Swapping from taking a component to taking component props because it makes it significantly easier to pass along the  `onEditDashboardClick` prop to trigger editing, which is necessary because a bunch of that logic is baked into this component. I don't think we need the empty state to be infinitely customizable, so this seems ok. Let me know if you disagree.

Reply:
Totally works for me - I tend to think of the empty state as the individual pieces anyway (title, description, actions)